bentoml/cli/__init__.py

```diff
@@ -152,7 +235,44 @@ def serve(port, archive_path=archive_path):
         help="Number of workers will start for the gunicorn server",
     )
     @click.option("--timeout", type=click.INT, default=None)
-    def serve_gunicorn(port, workers, timeout, archive_path=archive_path):
+    @click.option(
+        '--with-conda',
+        is_flag=True,
+        default=False,
+        help="Run API server on a Conda environment.",
+    )
+    def serve_gunicorn(
+        port, workers, timeout, archive_path=archive_path, with_conda=False
+    ):
+        if with_conda:
+            config = load_bentoml_config(archive_path)
+            metadata = config['metadata']
+            env_name = metadata['service_name'] + '_' + metadata['service_version']
+            pip_req = os.path.join(archive_path, 'requirements.txt')
+
+            subprocess.call(
+                'command -v conda >/dev/null 2>&1 || {{ echo >&2 "--with-conda '
+                'parameter requires conda but it\'s not installed."; exit 1; }} && '
+                'conda env update -n {env_name} -f {env_file} && '
+                'conda init bash && '
+                'eval "$(conda shell.bash hook)" && '
+                'conda activate {env_name} && '
+                '{{ [ -f {pip_req} ] && pip install -r {pip_req} || echo "no pip '
+                'dependencies."; }} &&'
+                'bentoml serve_gunicorn {archive_path} -p {port} -w {workers} '
```

Comment:
Is there a way to get all the CLI args from `click`? this makes it harder for us to add CLI args because we always need to remember to add it here - it's fine if there's no easy way to do it

Reply:
Yes, I believe there must be a way to get it, but while reading this doc http://click.palletsprojects.com/en/7.x/api/ I haven't found anything useful for this...

Reply:
After a little digging into the source code, I find `click` doesn't preserve the raw parameters once parsed them. Think  we can either clone and patch `click` or submit an issue to `click`, both of which seem not easy ways.